### PR TITLE
Add tests for expr.ServiceExpr.Error()

### PR DESCRIPTION
**Pull request by tchssk**
#1423

### CODE REVIEW

File: expr/service_test.go [hunk 1]
```diff
@@ -43,6 +43,42 @@ func TestServiceExprMethod(t *testing.T) {
...
+		expected *ErrorExpr
+	}{
+		"exist": {
+			name:     "foo",
+			expected: errorFoo,
+		},
+		"not exist": {
+			name:     "baz",
+			expected: nil,
+		},
```
Comment (nitinmohan87) [inline expr/service_test.go]: Would it be possible to add a test case where the `Root` expression has an error with the given `name` and `service.Error()` returns the correct error expression? Right now, your test cases does not cover this condition - https://github.com/goadesign/goa/blob/v2/expr/service.go#L70

File: expr/service_test.go [hunk 2]
```diff
@@ -43,6 +43,42 @@ func TestServiceExprMethod(t *testing.T) {
...
+	}
+
+	for k, tc := range cases {
```
Comment (nitinmohan87) [inline expr/service_test.go]: Let's start using `t.Run` for new tests. This we don't need to print `k` in `t.Errorf`.
```go
for k, tc := range cases {
  t.Run(k, func(t *testing.T) {
    ...
    if actual := s.Error(tc.name); actual != tc.expected {
      t.Errorf("invalid error expression: got %#v, expected %#v", actual, tc.expected)
    }
  })
}
```

File: expr/service_test.go (hunk 1 above)
Comment (tchssk) [reply expr/service_test.go]: Woops, https://github.com/goadesign/goa/pull/2024/commits/7d3a3363dda4cd999c1c795bc9e2d556e4d7368c adds the test.

File: expr/service_test.go (hunk 2 above)
Comment (tchssk) [reply expr/service_test.go]: https://github.com/goadesign/goa/pull/2024/commits/e21a2308251d4044c2c6bd6106ed7a61992958ae

**Comment by nitinmohan87** [comment]
Looks great.. Thank you so much!

**Pull request merged**